packages/openapi-generator/src/options/options-per-service.ts

```diff
@@ -29,12 +42,73 @@ export interface ServiceOptions {
  */
 export async function getOriginalOptionsPerService(
   configPath: string | undefined
-): Promise<OptionsPerService> {
+): Promise<PartialOptionsPerService> {
   return configPath && existsSync(configPath)
     ? JSON.parse(await readFile(configPath, 'utf8'))
     : {};
 }
 
+/**
+ * Get the options per service for the list of services.
+ * If optionsPerServicePath is not given default values are used for the services.
+ * If optionsPerServicePath is given exisiting values for the services are.
+ * @param inputPaths Paths for the service spec files
+ * @param options Options of the generator
+ * @returns The parsed configuration for all services in relativeServicePaths.
+ */
+export async function getOptionsPerService(
+  inputPaths: AbsoluteAndRelativePath[],
+  options: ParsedGeneratorOptions
+): Promise<OptionsPerService> {
+  const originalOptionsPerService = await getOriginalOptionsPerService(
+    options.optionsPerService
+  );
+
+  const uniqueNameGenerator = new UniqueNameGenerator('-');
+  const nonUniqueFiles: string[] = [];
```

Comment:
Consider removing the negation from this name. `pathsForDuplicateServiceNames` (or something shorter ;) )